### ISSUE TITLE
INSERT .. ON DUPLICATE KEY IGNORE emulation using MERGE should support multi value inserts

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10461]
This kind of statement:

```sql
INSERT INTO t (a, b)
VALUES (1, 2), (3, 4)
ON DUPLICATE KEY IGNORE
```

Is currently not emulated correctly when emulating it using `MERGE`. Only the last row is effectively inserted, without any warnings:

```sql
MERGE INTO t
USING (SELECT 1)
ON (a = 3)
WHEN NOT MATCHED THEN INSERT (a, b) VALUES (3, 4)
```

Instead, we need to push the data into the `USING` clause, as if we were emulating `INSERT .. SELECT .. ON DUPLICATE KEY IGNORE`.

This affects:

- DB2
- Derby
- H2 (only since 3.14 via #10460)
- HSQLDB
- Oracle
- SQL Server

**Comment by lukaseder** [comment]
In a first step, we'll be using the existing infrastruction (e.g. `FieldMapsForInsert::insertSelect`), which produces a `UNION ALL` concatenation, rather than a `VALUES()` constructor, which would be more lenient, syntactically:

```sql
-- We'll produce this
MERGE INTO t
USING (
  SELECT *
  FROM (
    SELECT 1, 2
    UNION ALL
    SELECT 3, 4
  ) u (a, b) -- Necessary step in H2, because derived column lists aren't supported for USING
) u
ON (t.a = u.a)
WHEN NOT MATCHED THEN INSERT (a, b) VALUES (u.a, u.b)

-- Instead of this
MERGE INTO t
USING (
  VALUES (1, 2), (3, 4)
) u (a, b)
ON (t.a = u.a)
WHEN NOT MATCHED THEN INSERT (a, b) VALUES (u.a, u.b)
```

There are various `VALUES()` and derived column list related SQL generation improvement issues.

**Comment by lukaseder** [comment]
An H2 (and probably also Derby, Oracle) limitation is the fact that derived column lists aren't supported in the `USING` clause: https://github.com/h2database/h2database/issues/2782

**Comment by lukaseder** [comment]
Fixed for 3.14.0. Backport scheduled for 3.13.5 (#10462)
